deepfence_console/fetcher/fetcher-server.go

```diff
@@ -445,6 +445,90 @@ func checkOwaspDependencyDataDownloading() bool {
 	return true
 }
 
+func handleVulnerabilityFeedTarUpload(respWrite http.ResponseWriter, req *http.Request) {
+	defer req.Body.Close()
+	if req.Method != "POST" {
+		http.Error(respWrite, "Invalid request", http.StatusInternalServerError)
+		return
+	}
+
+	// Maximum upload of 150 MB files
+	req.ParseMultipartForm(150 << 20)
+
+	inputFilePtr, handler, errVal := req.FormFile("file")
+	if handler == nil {
+		errMsg := "Error while getting data from client "
+		http.Error(respWrite, errMsg, http.StatusInternalServerError)
+		return
+	}
+
+	// Create file
+	dst, err := os.Create(handler.Filename)
+	if err != nil {
+		http.Error(respWrite, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	defer dst.Close()
+
+	if errVal != nil {
+		errMsg := "Error while writing post data. " + errVal.Error()
+		http.Error(respWrite, errMsg, http.StatusInternalServerError)
+		return
+	}
+	defer inputFilePtr.Close()
+	_, copyErr := io.Copy(dst, inputFilePtr)
+
+	// remove the tar.gz file after extraction
+	defer os.Remove(handler.Filename)
+
+	if copyErr != nil {
+		errMsg := "Error while writing post data. " + copyErr.Error()
+		http.Error(respWrite, errMsg, http.StatusInternalServerError)
+		return
+	}
+	extractFolder := "/root/.cache/grype/db/3"
+	go func() {
+		dst.Sync()
+		err = extractTarFile(handler.Filename, extractFolder)
+		msg := "Complete"
+		if err != nil {
+			msg = "Error while extracting file: " + err.Error()
+		}
+		statusFile, statusFileErr := os.OpenFile(extractFolder+"/extract_status.txt", os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+		if statusFileErr == nil {
+			_, _ = statusFile.WriteString(msg)
+			statusFile.Close()
+		}
+		vulnerabilityDbUpdater.updateVulnerabilityDbListing()
+
+		// call mapper api to update grype db
+		updateVulnerabilityMapperDB(respWrite)
+	}()
+	_, err = fmt.Fprintf(respWrite, "vulnerability db updated")
+	if err != nil {
+		fmt.Println(handler.Filename, err)
+	}
+
+	respWrite.WriteHeader(http.StatusOK)
+}
+
+func updateVulnerabilityMapperDB(respWrite http.ResponseWriter) {
```

Comment:
`respWrite` should not be passed here since it is running in a goroutine.
Since we have already sent 200 response at line 512, we should not be writing anything to `respWrite` anymore I think. If there are any errors we can log to stderr.

Reply:
pushed pls. check